stages/org.osbuild.rhsm

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+"""
+Configure Red Hat Subscription Management (RHSM)
+
+The stage currently supports configuring only RHSM DNF plugins,
+specifically enabling and disabling them. In the future, this
+stage may be extended to configure also other aspects of RHSM.
+
+In case the stage is configured to enable/disable specific
+DNF plugins, it expects that the appropriate configuration files
+exist in the filesystem tree. Non-existence of the configuration
+files will make the stage fail.
+"""
+
+import sys
+import iniparse
+
+import osbuild.api
+
+
+SCHEMA = """
+"additionalProperties": false,
+"properties": {
+  "dnf-plugins": {
+    "type": "object",
+    "description": "RHSM DNF plugins configuration",
+    "properties": {
+      "product-id": {
+        "type": "object",
+        "description": "'product-id' DNF plugin configuration",
+        "properties": {
+          "enabled": {
+            "type": "boolean",
+            "description": "enablement state of the plugin"
+          }
+        }
+      },
+      "subscription-manager": {
+        "type": "object",
+        "description": "'subscription-manager' DNF plugin configuration",
+        "properties": {
+          "enabled": {
+            "type": "boolean",
+            "description": "enablement state of the plugin"
+          }
+        }
+      }
+    }
+  }
+}
+"""
+
+
+def configure_dnf_plugins(tree, dnf_plugins_options):
+    for plugin, plugin_options in dnf_plugins_options.items():
```

Comment:
If i read this correctly, it relies on the fact that the schema is enforced, to not allow people to configure other plugins, or set unexpected options. That is ok, I think, but we should be aware that we need to keep enforcing this, so we don't end up accidentally allowing people to re-use this stage to do things we don't expect (and would then have to somehow maintain backwards compatibility with).

Reply:
OK, I think you found a valid issue that I didn't cover. I'll have to disallow additional properties in the "dnf-plugins" object. Options are enforced on line #73, but I'll add the same restrictions to schema. I actually thought that the schema will enforce it, but I'm just too new to jsonschema and didn't realize that I have to disallow additional properties for each object.

Reply:
Ah, the way that works is indeed a bit counter-intuitive / bad (I would have very much preferred if allowing additional properties was opt-in, rather than opt-out).

Reply:
Yeah, exactly. I adjusted the schema to disallow additional properties. Hopefully I got it finally right...